app/features/analysis/cards/OneRM/OneRMView.js

```diff
@@ -30,21 +30,34 @@ class OneRM extends Component {
     }
 
     _render1rm(confidence) {
-        if (confidence >= this.props.minConfidence) {
+        if (this.props.chartData.length > 3) {
+            if (confidence >= this.props.minConfidence) {
+                return (
+                    <View>
+                        <Text style={styles.oneRMText}>e1RM: <Text style={{fontWeight: 'bold'}}>{this.props.e1rm}</Text> {this.props.metric}</Text>
+                        <Text style={{ textAlign: 'center', fontSize: 15 }}>@ <Text style={{ fontWeight: 'bold' }}> {this.props.velocity} m/s</Text></Text> 
+                        <OneRMChart confidenceHighEnough={true} data={this.props.chartData} />
+                        <Text style={styles.confidenceText}>Confidence: {this.props.confidence} %</Text>
+                    </View>
+                );
+            } else {
+                return (
+                    <View>
+                        <Text style={styles.errorText}>
+                            Confidence too low, please clean up or log more data.
+                        </Text>
+                        <OneRMChart confidenceHighEnough={false} data={this.props.chartData} />
+                    </View>
+                );
+            }
+        } else {
             return (
                 <View>
-                    <Text style={styles.oneRMText}>e1RM: <Text style={{fontWeight: 'bold'}}>{this.props.e1rm}</Text> {this.props.metric}</Text>
-                    <Text style={{ textAlign: 'center', fontSize: 15 }}>@ <Text style={{ fontWeight: 'bold' }}> {this.props.velocity} m/s</Text></Text> 
-                    <OneRMChart data={this.props.chartData} />
-                    <Text style={styles.confidenceText}>Confidence: {this.props.confidence} %</Text>
+                    <Text style={styles.errorText}>
+                        This exercise has no reps within the date range.
+                    </Text>
```

Comment:
- [ ] is this no reps within range, or just not enough? why are you checking for length 3?

Reply:
Not enough reps within the date range, and you need 3 reps to perform a regression (verified by Adam)